components/TextPostCard.tsx

```diff
@@ -0,0 +1,186 @@
+import { StyleSheet, TouchableOpacity, Modal } from "react-native";
+import { TextInput } from "react-native";
+import { Button } from "react-native";
+import { Dimensions } from "react-native";
+import { ScrollView } from "react-native";
+import { Animated } from "react-native";
+
+const windowHeight = Dimensions.get("window").height;
+
+interface Comment {
+  id: string;
+  text: string;
+  username: string;
+  profilePic: string;
+  date: string; // Fügen Sie diese Zeile hinzu, wenn sie noch nicht existiert
+}
+
+interface Props {
+  username: string;
+  profilePic: string;
+  date: string;
+  initialLikes?: number;
+  postContent: any;
+  style?: React.CSSProperties;
+}
+
+const TextPostCard: React.FC<Props> = (props) => {
+  const {
+    username,
+    postContent,
+    profilePic,
+    date,
+    initialLikes = 149999,
+    style,
+  } = props;
+  const [likes, setLikes] = useState(initialLikes);
+  const [isLiked, setIsLiked] = useState(false);
+  const [isCommentModalVisible, setCommentModalVisible] = useState(false); // Zustand für das Kommentar-Popup
+  const [commentText, setCommentText] = useState(""); // Zustand für den Kommentar-Text
+  const [comments, setComments] = useState([]); // Hält die Liste der Kommentare
+  const [modalY, setModalY] = useState(new Animated.Value(windowHeight));
+
+  const addComment = () => {
+    if (commentText.trim()) {
+      const newComment = {
+        id: Date.now().toString(),
+        text: commentText,
+        username: username,
+        profilePic: profilePic,
+      };
+      setComments([...comments, newComment]);
+      setCommentText("");
+    }
+  };
+
+  const formatLikes = (count: number): string => {
+    const roundToTenths = (num: number) => Math.floor(num * 10) / 10;
+
+    if (count >= 1000000) {
+      return roundToTenths(count / 1000000) + " M";
+    }
+    if (count >= 1000) {
+      return roundToTenths(count / 1000) + " T";
+    }
+    return count.toString();
+  };
+
+  const handleLikePress = () => {
+    const newLikes = isLiked ? Math.max(likes - 1, 0) : likes + 1;
+    setLikes(newLikes);
+    setIsLiked(!isLiked);
+  };
+
+  const openCommentModal = () => {
+    setCommentModalVisible(true);
+  };
+
+  const closeCommentModal = () => {
+    setCommentModalVisible(false);
+  };
+
+  const handleCommentSubmit = () => {
+    setCommentText("");
+    closeCommentModal();
+  };
+
+  return (
+    <View className="items-center mx-2.5 mb-5">
+      <View className="w-full bg-white rounded-full p-4 shadow-lg z-20 relative">
+        <View className="flex-row items-center justify-between">
+          <Image
+            source={{ uri: profilePic }}
+            className="w-10 h-10 rounded-full"
+          />
+          <View className="flex-1 ml-2">
+            <Text className="font-bold">{username}</Text>
+            <Text className="text-xs text-gray-500">{date}</Text>
+          </View>
+          <TouchableOpacity onPress={openCommentModal}>
+            <Ionicons
```

Comment:
hier die Icon Komponente verwenden, um Konsistenz zu wahren. Siehe Icons in der TabBottomBar.tsx